test/inductor/test_torchinductor.py

```diff
@@ -309,25 +309,19 @@ def upcast_fn(x):
             else:
                 return x
 
-        def get_original_lowp_dtype(example_inputs):
-            dtypes = [x.dtype for x in example_inputs if isinstance(x, torch.Tensor)]
-            dtype_set = set(dtypes)
-            return dtype_set.pop() if len(dtype_set) == 1 else torch.half
-
         ref_inputs = list(map(upcast_fn, example_inputs))
         ref_kwargs = {k: upcast_fn(v) for k, v in kwargs.items()}
-        if has_lowp_args:
-            original_lowp_dtype = get_original_lowp_dtype(example_inputs)
-            if hasattr(model, "to"):
-                model = model.to(torch.float)
+        if has_lowp_args and hasattr(model, "to"):
+            orig_model = copy.deepcopy(model)
+            model = model.to(torch.float)
 
     torch.manual_seed(0)
 
     correct = model(*ref_inputs, **ref_kwargs)
     # downcast the model back if needed
-    if reference_in_float and has_lowp_args:
-        if hasattr(model, "to"):
-            model = model.to(original_lowp_dtype)
+    if reference_in_float and has_lowp_args and hasattr(model, "to"):
+        assert orig_model is not None
+        model = orig_model
```

Comment:
NIT: I think this would be simpler if model was always the original model, then you could do
```
ref_model = model

if cond:
   ref_model = copy.deepcopy(model).to(torch.float)

correct = ref_mode(*ref_inputs, **ref_kwargs)
```